cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -172,6 +172,17 @@ func CreateStackedEtcdStaticPodManifestFile(client clientset.Interface, manifest
 		return nil
 	}
 
+	if features.Enabled(cfg.FeatureGates, features.EtcdLearnerMode) {
+		learnerID, err := etcdClient.GetMemberID(etcdPeerAddress)
```

Comment:
This looks good. 

Should we double-check the `IsLeaner` of the member here? (logically not needed.)

Reply:
We could either add a method `IsLearner` or ignore the error if the member is already promoted.
I'm getting the following error while trying to promote a already promoted member

```
{"level":"warn","ts":"2023-01-13T09:49:08.421Z","logger":"etcd-client","caller":"v3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc0003c2a80/localhost:2379","attempt":0,"error":"rpc error: code = FailedPrecondition desc = etcdserver: can only promote a learner member"}
Error: etcdserver: can only promote a learner member
```

Reply:
Added the validation inside the `MemberPromote` method